[DOC-7830] Add more details about finalization, migrations, and how to monitor progress (#18609)
cockroachdb/docs: src/current/v24.1/upgrade-cockroach-version.md

{% assign previous_latest_prod = site.data.releases | where_exp: "previous_latest_prod", "previous_latest_prod.major_version == previous_version" | where: "release_type", "Production" | sort: "release_date" | last %}
{% assign actual_latest_prod = site.data.releases | where: "major_version", site.versions["stable"] | where: "release_type", "Production" | sort: "release_date" | last %}

> {% capture previous_version_numeric %}{{ previous_version | strip_first: 'v'}}{% endcapture %}

Reviewer: ```suggestion
{% capture previous_version_numeric %}{{ previous_version | remove_first: 'v'}}{% endcapture %}
```

> {% capture major_version_numeric %}{% assign local_major_version = page.major_version %}{{local_major_version | strip_first: 'v' }}{% endcapture %}

Reviewer: Maybe this will work? The var isn't displaying a value currently.

```suggestion
{% capture major_version_numeric %}{% assign local_major_version = page.version.version %}{{local_major_version | remove_first: 'v' }}{% endcapture %}
```

## Step 6. Finish the upgrade

Because a finalized major-version upgrade cannot be rolled back, Cockroach Labs recommends that you monitor the stability and performance of your cluster with the upgraded binary for at least a day before deciding to finalize the upgrade.

{{site.data.alerts.callout_info}}
Finalization is required only when upgrading from {{ previous_version }}.x to {{ page.version.version }}. For upgrades within the {{ page.version.version }}.x series, skip this step.
{{site.data.alerts.end}}

> 1. If you disabled auto-finalization in [step 3](#step-3-decide-how-the-upgrade-will-be-finalized), monitor the stability and performance of your cluster for at least a day. If you decide to roll back the upgrade, repeat the [rolling restart procedure](#step-4-perform-the-rolling-upgrade) with the previous binary. Otherwise, you must re-enable upgrade finalization to complete the upgrade to {{ page.version.version }}. Cockroach Labs recommends that you either finalize or roll back a major-version upgrade within a relative short period of time; running in a partially-upgraded state is not recommended.

Reviewer: Optional nit

```suggestion
1. If you disabled auto-finalization in [step 3](#step-3-decide-how-the-upgrade-will-be-finalized), monitor the stability and performance of your cluster for at least a day. If you decide to roll back the upgrade, repeat the [rolling restart procedure](#step-4-perform-the-rolling-upgrade) with the previous binary. Otherwise, perform the following steps to re-enable upgrade finalization to complete the upgrade to {{ page.version.version }}. Cockroach Labs recommends that you either finalize or roll back a major-version upgrade within a relative short period of time; running in a partially-upgraded state is not recommended.
```

    > RESET CLUSTER SETTING cluster.preserve_downgrade_option;
    ~~~

>     A series of migration jobs runs to enable certain types of features and changes in the new major version that cannot be rolled back. These include changes to system schemas, indexes, and descriptors, and enabling certain types of improvements and new features. Until the upgrade is finalized, these features and functions will not be available and the command `SHOW CLUSTER SETTING version` will return `{{ previous_version_numeric }}`.

Reviewer: `{{ previous_version_numeric }}` is still displaying `v23.2` and not `23.2`. If it needs to be the latter, then I figure the variable has to be fixed somehow?

Author: Here, the problem was that the variable I wanted is actually `page.version.version` on these pages, despite being `page.major_version` on normal pages.

Reviewer: It wasn't fixed by replacing `strip_first` with `remove_first`? Anyway, seems to be working now.

>     You can monitor the process of the migration in the DB Console [Jobs page]({% link {{ page.version.version }}/ui-jobs-page.md %}). Migration jobs have names in the format `{{ major_version_numeric }}-{migration-id}`. If a migration job fails or stalls, Cockroach Labs can use the migration ID to help diagnose and troubleshoot the problem. Each major version has different migration jobs with different IDs.

Reviewer: ``{{ major_version_numeric }}` isn't displaying any value on the page preview.
